specification/eventgrid/Azure.Messaging.EventGrid/client.tsp

```diff
@@ -1,8 +1,142 @@
 import "@azure-tools/typespec-client-generator-core";
 import "./main.tsp";
+import "@typespec/versioning";
 
 using Azure.ClientGenerator.Core;
+using Microsoft.EventGrid;
 
-@@access(Microsoft.EventGrid.publishCloudEvent, Access.internal, "python");
-@@access(Microsoft.EventGrid.publishCloudEvents, Access.internal, "python");
-@@access(Microsoft.EventGrid.receiveCloudEvents, Access.internal, "python");
+
+
+@TypeSpec.Versioning.useDependency(Azure.Core.Versions.v1_0_Preview_2)
+// NEED TO CHANGE THIS VERSION TO THE LATEST VERSION
+@TypeSpec.Versioning.useDependency(ServiceApiVersions.v2024_06_01)
+namespace Customizations{
+
+    // Python specific customizations
+    @client({ name: "EventGridPublisherClient", service: Microsoft.EventGrid}, "python")
+    interface Client1 {
+    send is publishCloudEvent;
+    sendEvents is publishCloudEvents;
+    }
+
+    @client({
+    name: "EventGridConsumerClient",
+    service: Microsoft.EventGrid
+    }, "python")
+    interface Client2 {
+    receive is receiveCloudEvents;
+    acknowledge is acknowledgeCloudEvents;
+    release is releaseCloudEvents;
+    reject is rejectCloudEvents;
+    renewLock is renewCloudEventLocks;
+    }
+
+    // Other Language specific customizations
+    @client({ name: "EventGridSenderClient", service: Microsoft.EventGrid}, "csharp")
+    interface ClientSender {
+    send is publishCloudEvent;
+    sendEvents is publishCloudEvents;
+    }
+
+    @client({
+    name: "EventGridReceiverClient",
+    service: Microsoft.EventGrid
+    }, "csharp")
+    interface ClientReceiver {
+    receive is receiveCloudEvents;
+    acknowledge is acknowledgeCloudEvents;
+    release is releaseCloudEvents;
+    reject is rejectCloudEvents;
+    renewLock is renewCloudEventLocks;
+    }
+
+    @client({ name: "EventGridSenderClient", service: Microsoft.EventGrid}, "js")
+    interface ClientSender1 {
+    send is publishCloudEvent;
+    sendEvents is publishCloudEvents;
+    }
+
+    @client({
```

Comment:
I do not think it is required. I have already split it in the custom layer. So, I am ok with keeping them as once client in the generated layer